CRM/Volunteer/Form/VolunteerSignUp.php

```diff
@@ -681,6 +681,10 @@ protected function setDestination() {
         // to that event form; otherwise, default to the vol opps page.
         if (count($this->_projects) === 1) {
           $project = current($this->_projects);
+          if (empty($project['entity_id'])) {
```

Comment:
I'm a little surprised that you're needing to check for the entity id since you're inside the 'event' case. Unless there's a bug somewhere else -- or you're generating your own links somehow -- the `dest` parameter should only be set to 'event' if you've arrived at the sign up form by clicking the "Volunteer Now" button from an Event Info page.